src/core/compiler.jl

```diff
@@ -35,6 +38,14 @@ Generate an observe expression.
 function generate_observe(observation, distribution)
     return esc(
         quote
+            # Sanity check.
+            isdist = if isa($(distribution), Vector)
```

Comment:
It's not immediately obvious from the way that the `distribution` variable is named that it is allowed to be either a `Distribution` or `Vector` thereof. Could we please make this clearer in the docstring for each method that has this variable as an argument?